src/main/java/io/jenkins/plugins/coverage/CoverageAction.java

```diff
@@ -0,0 +1,81 @@
+package io.jenkins.plugins.coverage;
+
+import hudson.model.Action;
+import hudson.model.Run;
+import io.jenkins.plugins.coverage.targets.CoverageResult;
+import jenkins.model.RunAction2;
+import jenkins.tasks.SimpleBuildStep;
+import org.kohsuke.stapler.StaplerProxy;
+
+import javax.annotation.CheckForNull;
+import java.lang.ref.WeakReference;
+import java.util.Collection;
+
+public class CoverageAction implements StaplerProxy, SimpleBuildStep.LastBuildAction, RunAction2 {
+
+    private Run<?, ?> owner;
+    private WeakReference<CoverageResult> report;
+
+
+    public CoverageAction(CoverageResult result) {
+        this.report = new WeakReference<>(result);
+    }
+
+    @Override
+    public Object getTarget() {
+        return getResult();
+    }
+
+    @Override
+    public Collection<? extends Action> getProjectActions() {
+        return null;
+    }
+
+    @CheckForNull
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @CheckForNull
+    @Override
+    public String getDisplayName() {
+        return "coverage report";
```

Comment:
I might be wrong, but I think this string shows up in the UI - if so, it should be made localizable